annotated.go

```diff
@@ -426,12 +423,16 @@ func (rt resultTagsAnnotation) results(ann *annotated) (
 	// if there's no Out struct among the return types, there was no As annotation applied
 	// just replace original result types with an Out struct and apply tags
 	var (
-		newOut       outStructInfo
-		existingOuts []reflect.Type
+		newOut              outStructInfo
+		existingOuts        []reflect.Type
+		existingOutsMapping = make(map[int][]reflect.Type)
 	)
 
 	newOut.Fields = []reflect.StructField{_outAnnotationField}
 	newOut.Offsets = []int{}
+	// to prevent duplicate applying of the same tag to the same type, it is kept in the following format
+	// {"foo": {"fmt.Stringer": struct{}{}, "myStringer": struct{}{}}, "bar": {"fmt.Stringer": struct{}{}}}
+	retaggedTypeMap := map[string]map[string]struct{}{}
```

Comment:
Can we validate whether or not a given set of ResultTags annotations will provide conflicting types to the container before we try to build out an actual constructor? (in apply() method) If we validate it beforehand, we won't have to worry about this at all here.